File: cpp/gtest/NorwegianBlueParrot.cpp
#include "NorwegianBlueParrot.h"
#include <algorithm>


NorwegianBlueParrot::NorwegianBlueParrot(double voltage, bool isNailed) : Parrot(), voltage(voltage), isNailed(isNailed)
{

}

double NorwegianBlueParrot::getSpeed() {
	return (isNailed) ? 0 : getBaseSpeed(voltage);
}


double NorwegianBlueParrot::getBaseSpeed(double current_voltage) {
	return std::min(24.0, current_voltage * baseSpeed);
}

File: TypeScript/jest/src/parrots/norwegian-blue-parrot.ts
import AbstractParrot from '../abstract-parrot';

export default class NorwegianBlueParrot extends AbstractParrot{

    private readonly MAX_SPEED = 24;

    private readonly voltage: number;

    private readonly isNailed: boolean;

    constructor(voltage: number, isNailed: boolean) {
        super();
        this.voltage = voltage;
        this.isNailed = isNailed;
    }

    getSpeed(): number {
        return (this.isNailed) ? 0 : this.getBaseSpeedWithVoltage(this.voltage);
    }

    private getBaseSpeedWithVoltage(voltage: number): number {
        return Math.min(this.MAX_SPEED, voltage * this.getBaseSpeed());
    }
}
